contracts/ValidatorFactory.sol

```diff
@@ -0,0 +1,123 @@
+/// @notice It allows to create an remove validator proxy that are used to StakeManager
+contract ValidatorFactory is
+    Initializable,
+    OwnableUpgradeable,
+    UUPSUpgradeable,
+    ValidatorFactoryStorage
+{
+    // ====================================================================
+    // =========================== MODIFIERS ==============================
+    // ====================================================================
+
+    /// @notice Check if the msg.sender is the owner.
+    modifier isOwner() {
+        require(owner() == msg.sender, "Permission not found");
+        _;
+    }
+
+    /// @notice Check if the operator contract is the msg.sender.
+    modifier isOperator() {
+        require(
+            state.operator == msg.sender,
+            "Caller is not the operator contract"
+        );
+        _;
+    }
+
+    // ====================================================================
+    // =========================== FUNCTIONS ==============================
+    // ====================================================================
+
+    /// @notice Initialize the NodeOperator contract.
+    function initialize(address _validatorImplementation) public initializer {
+        __Ownable_init();
+        state.validatorImplementation = _validatorImplementation;
+    }
+
+    /// @notice Deploy a new validator contract
+    /// @return return the address of the new validator contract deployed
+    function create() external isOperator returns (address) {
+        require(state.operator != address(0), "Operator contract not set");
+
+        // create a new validator proxy
+        address proxy = address(
+            new ValidatorProxy(owner(), state.validatorImplementation)
+        );
+
+        // set the operator address.
+        (bool success, ) = proxy.call(
+            abi.encodeWithSignature("setOperator(address)", state.operator)
+        );
+        require(success, "Set operator fails");
+
+        validators.push(proxy);
+
+        emit CreateValidator(proxy);
+        return proxy;
+    }
+
+    /// @notice Remove a validator proxy from the list.
+    function remove(address _validatorProxy) external isOperator {
+        for (uint256 idx = 0; idx < validators.length; idx++) {
+            if (_validatorProxy == validators[idx]) {
+                validators[idx] = validators[validators.length - 1];
+                validators.pop();
+            }
+        }
+    }
+
+    /// @notice Get validators contracts.
+    /// @return return a list of deployed validator contracts.
+    function getValidators() public view returns (address[] memory) {
+        return validators;
+    }
+
+    /// @notice Implement _authorizeUpgrade from UUPSUpgradeable contract to make the contract upgradable.
+    /// @param newImplementation new contract implementation address.
+    function _authorizeUpgrade(address newImplementation)
+        internal
+        override
+        isOwner
+    {}
```

Comment:
TODO?

Reply:
No this one is to upgrade the contract is does not require any implementation only adding the modifier isOwner to make sure only him can upgrade.